hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieArchivedTimeline.java

```diff
@@ -143,15 +149,19 @@ public HoodieArchivedTimeline reload() {
   }
 
   private HoodieInstant readCommit(GenericRecord record, boolean loadDetails) {
-    final String instantTime  = record.get(HoodiePartitionMetadata.COMMIT_TIME_KEY).toString();
+    final String instantTime = record.get(HoodiePartitionMetadata.COMMIT_TIME_KEY).toString();
     final String action = record.get(ACTION_TYPE_KEY).toString();
+    String actionState = record.get(ACTION_STATE).toString();
     if (loadDetails) {
       getMetadataKey(action).map(key -> {
         Object actionData = record.get(key);
-        if (action.equals(HoodieTimeline.COMPACTION_ACTION)) {
-          this.readCommits.put(instantTime, HoodieAvroUtils.indexedRecordToBytes((IndexedRecord)actionData));
-        } else {
-          this.readCommits.put(instantTime, actionData.toString().getBytes(StandardCharsets.UTF_8));
+        // Only puts instant details if the instant is completed and there is action data
+        if (HoodieInstant.State.COMPLETED.toString().equals(actionState) && actionData != null) {
```

Comment:
looked at the code to ensure that archival will not move past the earlist inflight instant in timline. But just inacse someone wants to double check, leaving a comment here. if this assumption is wrong, this fix has to be reverted.

Reply:
Yeah, that's a good point.  The archival should not have inflight instant.  To avoid breaking changes, I used a different way of fixing the `readCommit`.